pytorch_lightning/utilities/auto_restart.py

```diff
@@ -318,31 +315,17 @@ def _wrap_generator_samplers(self) -> None:
             if isinstance(generator, Sampler):
                 continue
 
-            # used to handle a weird behaviour from PyTorch 1.6
-            # where the sampler is converted to a list_iterator
-            is_legacy = False
```

Comment:
@tchaton can you carefully review the changes to this file?

I had to keep the `is_legacy=True` codepath instead of the other way around. Which seems wrong since we want to keep the new codepath.

Perhaps is_legacy was misnamed?

Reply:
Yes, wrongly named. Your change seems good to me.